cmd/bosun/sched/sched.go

```diff
@@ -391,7 +396,18 @@ func (s *Schedule) MarshalGroups(T miniprofiler.Timer, filter string) (*StateGro
 				}
 				continue
 			}
-			if matches(s.Conf, a, v) {
+			if parsedExpr == nil {
```

Comment:
I feel like the logic here is a bit odd. I would like to see the `status[k] = v` bit only once, and continue of it doesn't match the filter. 
Something like `if filter and !filter.Matches(is) { continue}   status[k] = v`. If that makes any sense.